src/group_theory/monoid_localization.lean

```diff
@@ -266,25 +274,295 @@ show _ * _ = _, by rw [mul_inv_left, mul_one]
   f.mk' x ⟨x, H⟩ = 1 :=
 by convert mk'_self _ _; refl
 
-@[simp, to_additive] lemma mul_mk'_eq_mk'_of_mul (x₁ x₂) (y : S) :
+@[to_additive] lemma mul_mk'_eq_mk'_of_mul (x₁ x₂) (y : S) :
   f.1 x₁ * f.mk' x₂ y = f.mk' (x₁ * x₂) y :=
 by rw [←mk'_one, ←mk'_mul, one_mul]
 
-@[simp, to_additive] lemma mk'_mul_eq_mk'_of_mul (x₁ x₂) (y : S) :
+@[to_additive] lemma mk'_mul_eq_mk'_of_mul (x₁ x₂) (y : S) :
   f.mk' x₂ y * f.1 x₁ = f.mk' (x₁ * x₂) y :=
 by rw [mul_comm, mul_mk'_eq_mk'_of_mul]
 
-@[simp, to_additive] lemma mul_mk'_one_eq_mk' (x) (y : S) :
+@[to_additive] lemma mul_mk'_one_eq_mk' (x) (y : S) :
   f.1 x * f.mk' 1 y = f.mk' x y :=
 by rw [mul_mk'_eq_mk'_of_mul, mul_one]
 
-@[simp, to_additive] lemma mk'_mul_cancel_right (x : M) (y : S) :
+@[to_additive] lemma mk'_mul_cancel_right (x : M) (y : S) :
   f.mk' (x * y) y = f.1 x :=
 by rw [←mul_mk'_one_eq_mk', f.1.map_mul, mul_assoc, mul_mk'_one_eq_mk', mk'_self, mul_one]
 
-@[simp, to_additive] lemma mk'_mul_cancel_left (x) (y : S) :
+@[to_additive] lemma mk'_mul_cancel_left (x) (y : S) :
   f.mk' ((y : M) * x) y = f.1 x :=
 by rw [mul_comm, mk'_mul_cancel_right]
 
+variables {g : M →* P}
+
+@[to_additive] lemma comp_eq_of_eq {T : submonoid P} {Q : Type*} [comm_monoid Q]
+  (hg : ∀ y : S, g y ∈ T) (k : localization_map T Q)
+  {x y} (h : f.1 x = f.1 y) : k.1 (g x) = k.1 (g y) :=
+let ⟨c, hc⟩ := (f.4 _ _).1 h in (k.4 _ _).2
+  ⟨⟨g c, hg c⟩, show g _ * g _ = g _ * g _, by rw [←g.map_mul, hc, g.map_mul]⟩
+
+variables (hg : ∀ y : S, is_unit (g y))
+include hg
```

Comment:
Do you need the include statement? Apart from `eq_of_eq` (where you could add it to the list of hypotheses explicitly) it seems that all the types mention `hg`, so Lean would include `hg` automatically.